src/olympia/devhub/views.py

```diff
@@ -1858,7 +1858,12 @@ def get_promoted_subscription_or_404(addon):
         raise http.Http404()
 
     qs = PromotedSubscription.objects.select_related('promoted_addon__addon')
-    return get_object_or_404(qs, promoted_addon__addon=addon)
+    subscription = get_object_or_404(qs, promoted_addon__addon=addon)
```

Comment:
for an efficiency gain we could filter by subscriptions that are `stripe_checkout_completed` and not `cancelled_at` directly here (would need `Q` objects I would have thought).  Would make it a little less readable though.

Reply:
This will be needed anyway if there's a future where we have multiple `PromotedSubscription` (one is cancelled, and later a new one starts).

Reply:
I think that wouldn't work for the `cancel` endpoint, which wouldn't have `stripe_checkout_completed` set yet.

> This will be needed anyway if there's a future where we have multiple PromotedSubscription (one is cancelled, and later a new one starts).

I agree. This is something I'd like to investigate in the near future, but I am not sure how to do that so I'll start by creating an issue.

Reply:
see: https://github.com/mozilla/addons-server/issues/15967

Reply:
> I think that wouldn't work for the cancel endpoint, which wouldn't have stripe_checkout_completed

I was meaning: replicate the exact logic, so it would be not stripe_checkout_completed or stripe_checkout_completed and not cancelled_at.